### PR TITLE
Merge release 0.22 into gold/2021

### PR DESCRIPTION
**Pull request by diptorupd**
Merges `main` containing 0.22.0 tag into gold/2021

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
This PR looks good to me. Whether it can still be picked up for the release is up to @xaleryb

**Review by oleksandr-pavlyk — DISMISSED** [review]

**Comment by ekomarova** [comment]
I'm going to fix the problem with wheels before merging this

**Comment by ekomarova** [comment]
@diptorupd  conda linux tests end with code 0 but in fact they didn't even start because of the following problem:
```
python -c 'import dpctl; print(" ".join([dev.backend.name+":"+dev.device_type.name for dev in dpctl.get_devices() if dev.device_type.name in ["cpu","gpu"]]))'
      from ._sycl_device import SyclDevice, SyclDeviceCreationError
  ImportError: dpctl/libDPCTLSyclInterface.so.0: undefined symbol: _ZN4sycl3_V17handler19getMaxWorkGroups_v2Ev
```

Speaking of wheels there are several tests failing:
```
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-device-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-device-shape1] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-host-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-host-shape1] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-shared-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int32-shared-shape1] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-device-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-device-shape1] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-host-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-host-shape1] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-shared-100] - Failed: Failed to compile.
FAILED tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators[parfor_ipow-int64-shared-shape1] - Failed: Failed to compile.
```

**Review by xaleryb — APPROVED** [review]

**Pull request merged**
